apm-sniffer/apm-sdk-plugin/solr-7.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/solr/search/QueryStageTracerInterceptor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.solr.search;
+
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.ContextSnapshot;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+import org.apache.skywalking.apm.plugin.solr.commons.NamesMap;
+import org.apache.solr.handler.component.ResponseBuilder;
+
+import java.lang.reflect.Method;
+
+import static org.apache.skywalking.apm.plugin.solr.commons.Constants.OPER_STAGE_PREF;
+
+public class QueryStageTracerInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        if (method.getName().equals("distributedProcess")) {
+            ContextSnapshot snapshot = null;
+            if (ContextManager.isActive()) {
+                snapshot = ContextManager.capture();
+            }
+
+            ResponseBuilder rb = (ResponseBuilder) allArguments[0];
+            AbstractSpan span = ContextManager.createLocalSpan(OPER_STAGE_PREF + NamesMap.getStagName(rb.stage));
+            span.setComponent(ComponentsDefine.SOLR);
+
+            if (snapshot != null) {
+                ContextManager.continued(snapshot);
+            }
+        }
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                              Object ret) throws Throwable {
+        if (method.getName().equals("finishStage")) {
```

Comment:
i'm a bit confused about `if (method.getName().equals("distributedProcess"))` in before method and `if (method.getName().equals("finishStage")) {` in after method

this two method name should be same.but this logic judgment is not

Reply:
This span start at `distributedProcess()` and stop at `finishStage()`. 
But I'm pretty sure they always come in pairs.